compiler/passes/createTaskFunctions.cpp

```diff
@@ -211,24 +315,56 @@ void pruneThisArg(Symbol* parent, SymbolMap& uses) {
 }
 
 static void
-addVarsToFormals(FnSymbol* fn, SymbolMap& vars) {
+addVarsToFormalsActuals(FnSymbol* fn, SymbolMap& vars,
+                        CallExpr* call, bool isCoforall)
+{
+  Expr *redRef1 = NULL, *redRef2 = NULL;
   form_Map(SymbolMapElem, e, vars) {
       Symbol* sym = e->key;
       if (e->value != markPruned) {
         SET_LINENO(sym);
-        IntentTag argTag = INTENT_BLANK;
-        if (ArgSymbol* tiMarker = toArgSymbol(e->value))
-          argTag = tiMarker->intent;
-        else
-          INT_ASSERT(e->value == markUnspecified);
-        ArgSymbol* arg = new ArgSymbol(argTag, sym->name, sym->type);
-        if (ArgSymbol* symArg = toArgSymbol(sym))
-          if (symArg->hasFlag(FLAG_MARKED_GENERIC))
-            arg->addFlag(FLAG_MARKED_GENERIC);
-        fn->insertFormalAtTail(new DefExpr(arg));
-        e->value = arg;  // aka vars->put(sym, arg);
+        ArgSymbol* newFormal = NULL;
+        Symbol*    newActual = NULL;
+        Symbol*    symReplace = NULL;
+
+        if (TypeSymbol* reduceType = toTypeSymbol(e->value)) {
```

Comment:
Can you add a comment about why e->value being a type symbol means that we have a reduction type?